devlib/module/sched.py

```diff
@@ -309,13 +309,23 @@ class SchedProcFSData(SchedProcFSNode):
     Root class for creating & storing SchedProcFSNode instances
     """
     _read_depth = 6
-    sched_domain_root = '/proc/sys/kernel/sched_domain'
+
+    @classmethod
+    def get_data_root(cls, target):
+        # Location differs depending on kernel version
+        for root in ['/sys/kernel/debug/sched/domains/', '/proc/sys/kernel/sched_domain']:
+            if target.file_exists(root):
+                return root
+
+        return None
```

Comment:
Looks like this should raise an exception. Returning None from a function that usually returns a valid value is one of the greatest footgun, which is already being triggered in that very PR:
```
        if path is None:
            path = SchedProcFSData.get_data_root(target)

        procfs = target.read_tree_values(path, depth=self._read_depth)
```

There is not guarantee that `path is not None`, which will result in a shell call:
```
read_tree_values "''" 1
```
which will probably fail with a non obvious error.

In the absence of good static typing along with something like Rust's Option<T>, exceptions are much more desirable. They have basically the same typical behavior (i.e. bypassing the rest of the code and "returning" early), and they can be caught in the odd case where you want to turn it into a boolean like in `def available`.